floodviz/test/hydrograph_utils_tests.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+
+class TestReqHydroData(unittest.TestCase):
+
+  def setUp(self):
+    self.prefix = 'https://waterservices.usgs.gov/nwis/'
+    self.H_START_DT = '2008-05-20'
+    self.H_END_DT = '2008-07-05'
+    self.sites = ['05463500', '05471050', '05420680', '05479000', '05484000', '05481000', '05486000', '05421000', '05485500',
+              '05455100', '05470500', '05451500', '05458000', '05471000', '05462000', '05457700', '05458500', '05470000',
+              '05484500', '05481300', '05464220', '05458900', '05485605', '05463000', '05471200', '05476750', '05411850',
+              '05454220', '05481950', '05416900', '05464500', '05487470']   
+
+  def test_bad_list(self):
+    self.assertEqual(req_hydrodata(['x'], self.H_START_DT, self.H_END_DT, self.prefix), [])
+    self.assertEqual(req_hydrodata([123456, 100], self.H_START_DT, self.H_END_DT, self.prefix), [])
+    self.assertEqual(req_hydrodata([], self.H_START_DT, self.H_END_DT, self.prefix), [])
+
+  def test_bad_url(self):
+    self.assertEqual(req_hydrodata(self.sites, self.H_START_DT, self.H_END_DT, 'badurl.noob' ), [])
+    self.assertEqual(req_hydrodata(self.sites, self.H_START_DT, self.H_END_DT, ''), [])
+
+  def test_valid_data(self):
+    self.assertTrue(req_hydrodata(self.sites, self.H_START_DT, self.H_END_DT, self.prefix))
```

Comment:
As for the last test you mentioned (passing in an empty list and expecting a returned empty list), wouldn't my test_empty_data in TestParseHydroData work for this?